src/ppx/sqlexpr_parser.ml

```diff
@@ -0,0 +1,92 @@
+type typ = Int | Int32 | Int64 | Float | Text | Blob | Bool | Any
+type input = typ * bool
+type output = string * typ * bool
+
+let str2typ = function
+  | "d" -> Int
+  | "l" -> Int32
+  | "L" -> Int64
+  | "f" -> Float
+  | "s" -> Text
+  | "S" -> Blob
+  | "b" -> Bool
+  | "a" -> Any
+  | _ -> failwith "Invalid type"
+
+let typ2str = function
+  | Int -> "int"
+  | Int32 -> "int32"
+  | Int64 -> "int64"
+  | Float -> "float"
+  | Text -> "text"
+  | Blob -> "blob"
+  | Bool -> "bool"
+  | Any -> "any"
+
+let in_type2str ((typ, optional) : input) =
+  let typ = typ2str typ in
+  if optional then "maybe_" ^ typ else typ
+
+let out_type2str ((_, typ, optional) : output) =
+  in_type2str (typ, optional)
+
+let parse str =
+  (* grievous hack to escape everything within quotes *)
+  (* what about ignoring \' or \" ?       " *)
+  (* manually escape "" because {| ... |} notation breaks syntax highlighting *)
+  let escrgx = Re_pcre.regexp "('[^']*')|(\"[^\"]*\")" in
+  let esc_list = ref [] in
+  let esc_str = "<SQLEXPR_PRESERVED>" in
+  let esc_subst substrings =
+    let mtch = Re.get substrings 0 in
+    esc_list := mtch :: !esc_list;
+    esc_str in
+
+  let escaped = Re.replace ~f:esc_subst escrgx str in
+  esc_list := List.rev !esc_list;
+
+  (* logic to extract inputs and outputs *)
+  let inrgx = Re_pcre.regexp {|%([dlLfsSba])(\?)?|} in
+  let outrgx = Re_pcre.regexp {|@([dlLfsSba])(\?)?\{((\w|\.)+)\}|} in
```

Comment:
Indeed. Fixed this and all the other remaining issues, sent another PR that goes into your ppx branch: https://github.com/mfp/ocaml-sqlexpr/pull/10